Parametrize demo edge script

BaptisteGi opened the pull request:
Changes:
- Introduced new parameters for the command
```
infrahubctl run models/infrastructure_edge.py profile="large"
infrahubctl run models/infrastructure_edge.py num_sites=10 num_device_per_site=14
infrahubctl run models/infrastructure_edge.py has_bgp_mesh=False has_branch=False
```
- Introduced methods to leverage these parameters during object generation
- Had to increase prefixes size to fit all those potential 
- Removed the `Create iBGP Sessions within the Site` part as it's sort of duplicated with the mesh creation
- [ ] Improve performance? => Will do it in a different PR
- [ ] Properly leverage protocol feature from the SDK

BaptisteGi committed:
Introduce config object

BaptisteGi committed:
Introduce config object

BaptisteGi committed:
Create 200 sites

BaptisteGi committed:
Create 200 sites

BaptisteGi commented on models/infrastructure_edge.py:
```diff
@@ -1554,12 +1901,13 @@ async def generate_continents_countries(client: InfrahubClient, log: logging.Log
-    groups = await client.filters(branch=branch, kind="CoreUserGroup", name__value="Administrators")
-    store.set(key=groups[0].name, node=groups[0])
+    # FIXME: For some reason this part doesn't work
+    # groups = await client.filters(branch=branch, kind="CoreUserGroup", name__value="Administrators")
```
It doesn't work for me for some reason ..

dgarros commented on models/infrastructure_edge.py:
```diff
@@ -185,15 +284,74 @@ class Vlan(BaseModel):
 
+# Chat GPT came up with the most cliche contact names here
 SITES = [
+    Site(name="sin", country="Singapore", city="Singapore", contact="Siti Tan"),
```
I would prefer to keep the list of sites short to keep this file as easy to read as possible
the `site_generator` doesn't need that many sites, it will automatically increase the ID number if we need to generate more than 10 sites.

```
    site_names_generator(nbr_site=5)
        result >> ["atl1", "ord1", "jfk1", "den1", "dfw1"]

    site_names_generator(nbr_site=12)
        result >> ["atl1", "ord1", "jfk1", "den1", "dfw1", "iad1", "bkk1", "sfo1", "iah1", "mco1", "atl2", "ord2"]
```

BaptisteGi committed:
Support branch and bgp settings

BaptisteGi committed:
Merge branch 'develop' into bgi-parametrize-demo-script-4233

BaptisteGi committed:
Masta comito: site design, improve device p2p, fix account function, fix site list

BaptisteGi committed:
Refine values for profile

BaptisteGi committed:
Fix unit test with management IPs

BaptisteGi committed:
Adjust prefixes values in e2e tests

BaptisteGi committed:
Fix further e2e tests

BaptisteGi committed:
Remove unwanted changes around account

BaptisteGi committed:
Another round of fixing

ogenstad committed:
Remove GetListMixin

dgarros committed:
Add event service

dependabot[bot] committed:
Bump actions/setup-python from 4 to 5

Bumps [actions/setup-python](https://github.com/actions/setup-python) from 4 to 5.
- [Release notes](https://github.com/actions/setup-python/releases)
- [Commits](https://github.com/actions/setup-python/compare/v4...v5)

---
updated-dependencies:
- dependency-name: actions/setup-python
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dgarros committed:
Generate random INFRAHUB_DB_BACKUP_PORT during backend-benchmark

BaptisteGi committed:
Fix various issues around intefaces and platform

BaptisteGi committed:
Merge branch 'develop' into bgi-parametrize-demo-script-4233

BaptisteGi committed:
Adjust following rebase

BaptisteGi committed:
Fix e2e tests

fatih-acar committed:
fix(ci): disable NATS jobs

Signed-off-by: Fatih Acar <fatih@opsmill.com>

gmazoyer committed:
IFC-673 Add super admin permission (#4458)

This permission allows to bypass all others. It will be verified first.
If a user has this permission, other permission checkers won't run
which implies other permissions won't be evaluated.

bilalabbad committed:
Add SSO in Infrahub UI (#4497)

added sso buttons on sign in page
handle sso auth callback in UI

BaptisteGi committed:
Merge branch 'develop' into bgi-parametrize-demo-script-4233

BaptisteGi committed:
Merge branch 'develop' into bgi-parametrize-demo-script-4233
